src/app/core/services/label.service.ts

```diff
@@ -70,36 +95,92 @@ export class LabelService {
   }
 
   /**
-   * Stores the json data from Github api into the list of arrays in this service
-   * @param labels: the json data of the label
+   * Returns an array of Preset Labels.
    */
-  private populateLabelLists(labels: Array<{}>): void {
+  private getRequiredLabels(): Label[] {
+    const requiredLabels: Label[] = [];
 
-    for (const label of labels) {
-      // Get the name and color of each label and store them into the service's array list
-      const labelName = String(label['name']).split('.');
-      const labelType = labelName[0];
-      const labelValue = labelName[1];
-      const labelColor = String(label['color']);
-
-      // Check for duplicate labels
-      if (this.labelColorMap.has(labelValue)) {
-        continue;
+    for (const category of Object.keys(this.REQUIRED_LABELS)) {
+      for (const labels of Object.values(this.REQUIRED_LABELS[category])) {
+        requiredLabels.push(labels as Label);
       }
+    }
+
+    return requiredLabels;
+  }
+
+  /**
+   * Stores the json data from Github api into the list of arrays in this service
+   * @param labels: the json data of the label
+   * @param expectedLabels: An array of labels that are required by the application.
+   */
+  private storeLabelData(labels: Array<{}>, expectedLabels: Label[]): void {
```

Comment:
See if the header comment and the function name can be improved to reflect what the function does.

Reply:
@damithc 
Would the following be better?
```typescript
 /**
   * Compares and stores labels represented by a JSON Object with those required by
   * the application and updates the remote labels.
   * @param labels: JSON data representing labels.
   * @param expectedLabels: An array of labels that are required by the application.
   */
  private compareAndStoreLabels(labels: Array<{}>, expectedLabels: Label[]): void {
    // ...
  }
```

Reply:
Something like this?
```js
/**
   * Ensures that the repo has the expected labels.
   * Compares the actual labels in the repo with expected labels. If an expected label is missing,
   * it is added to the repo. If the expected label exists but the label color is not as expected,
   * the color is updated. Does not delete actual labels that do not match expected labels.
   * i.e., the repo might have more labels than the expected labels after this operation.
   * @param labels: JSON data representing labels in the repo.
   * @param expectedLabels: expected labels.
   */
  private ensureRepoHasExpectedLabels(actualLabels: Array<{}>, expectedLabels: Label[]): void {
    // ...
  }
```

Reply:
A query, Does the method definition need to capture the "saving labels for our app to utilise"? Or as a rule of thumb we just state the main task that it carries out?

Reply:
That information not relevant to the method. The less the method know, the better. The method simply ensures the given labels are in the repo, no less, no more. It does not need to know why the labels need to be in the repo. In fact, there can be many reasons, although at the moment there is only one known reason.